File: hospitalManagement/Repositories.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hospitalManagement
{
    internal class Repositories : IBasicActivities<Repository>
    {
        //Field
        private List<Repository> repositoryList;
        private long count;






        // Infor field
        //  Management field
        // Dynamic field
        // Properties


        internal List<Repository> RepositoryList { get => repositoryList; set => repositoryList = value; }
        public long Count { get => count; set => count = value; }

        // Constructors
        public Repositories()
        {
            repositoryList = new List<Repository>();
            this.Count = 0;
        }
        public Repositories(List<Repository> repositoryList, long count)
        {
            this.repositoryList = repositoryList;
            this.count = count;
        }

        // Destructors
        // Methods
        // in, output
        // General method
        // Other method

        public void AddItem()
        {
            Repository repository = new Repository();
            Console.WriteLine("New repository");
            repository.Input();
            repositoryList.Add(repository);
            Console.WriteLine("Done!");

        }

        public void AddItem(Repository value)
        {
            Console.WriteLine("New repository");
            repositoryList.Add(value);
            Console.WriteLine("Done!");
        }
        public void ShowInformation()
        {
            Console.WriteLine("Show information of all repositories");

            repositoryList.ForEach(value =>
            {
                value.Output();
                Console.WriteLine();
            });
            Console.WriteLine("Done!");

        }

        public Repository UpdateItem(string id)
        {
            Console.WriteLine("Update the repository");

            Repository res = null;
            repositoryList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    value.Input();
                    res = value;
                }

            });
            if (res == null)
            {
                Console.WriteLine($"Not found the repository with id: {id}");
            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public bool RemoveItem(string id)
        {
            Console.WriteLine("Remove the repository");

            bool res = false;
            repositoryList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    repositoryList.Remove(value);
                    res = true;
                }
            });
            if (res == false)
            {
                Console.WriteLine($"Not found repository with id: {id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public Repository FindItem(string id)
        {
            Console.WriteLine("Find the repository");

            Repository res = null;
            repositoryList.ForEach(value =>
            {
                if (value.Id == id)
                {
                    res = value;
                }
            });
            if (res == null)
            {
                Console.WriteLine($"Not found repository with id: {id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public bool Clear()
        {
            Console.WriteLine("Clear all repositories");

            repositoryList.Clear();
            if (repositoryList.Count > 0)
            {
                Console.WriteLine("Failure!");
                return false;
            }
            this.Count = 0;
            Console.WriteLine("Done!");

            return true;

        }
        // Overriding
    }
}

File: src/hospitalManagement/Repositories.java
package hospitalManagement;

import java.util.ArrayList;

public class Repositories implements IBasicActivities<Repository> {
	//Field
    private ArrayList<Repository> repositoryList;
    private long count;

    // Infor field
    //  Management field
    // Dynamic field
    // Properties


    

    public ArrayList<Repository> getRepositoryList() {
		return repositoryList;
	}
	public void setRepositoryList(ArrayList<Repository> repositoryList) {
		this.repositoryList = repositoryList;
	}
	public long getCount() {
		return count;
	}
	public void setCount(long count) {
		this.count = count;
	}
	// Constructors
    public Repositories()
    {
        repositoryList = new ArrayList<Repository>();
        this.count = 0;
    }
    public Repositories(ArrayList<Repository> repositoryList, long count)
    {
        this.repositoryList = repositoryList;
        this.count = count;
    }

    // Destructors
    // Methods
    // in, output
    // General method
    // Other method

    public void AddItem()
    {
        Repository repository = new Repository();
        System.out.println("New repository");
        repository.Input();
        repositoryList.add(repository);
        this.count++;
        System.out.println("Done!");
    }

    public void AddItem(Repository value)
    {
        System.out.println("New repository");
        repositoryList.add(value);
        this.count++;
        System.out.println("Done!");
    }
    public void ShowInformation()
    {
        System.out.println("-------------------------------------------------------------------");

        System.out.println("Show information of all repositories");

        repositoryList.forEach(value ->
        {
            value.Output();
            System.out.println();
        });
        System.out.println("Done!");
        System.out.println("-------------------------------------------------------------------");

    }

    public Repository UpdateItem(String id)
    {
        System.out.println("Update the repository");

        Repository  res = null;
        for (Repository rep : repositoryList) {
			if( rep.getId() == id)
			{
				rep.Input();
				res = rep;
			}
		}
        if (res == null)
        {
            System.out.format("\nNot found the repository with id: $s\n", id);
        }
        else
        {
            System.out.println("Done!");
        }
        return res;
    }

    public boolean RemoveItem(String id)
    {
        System.out.println("Remove the repository");

        boolean res = false;
        for (Repository rep : repositoryList) {
			if(rep.getId()== id)
			{
				repositoryList.remove(rep);
				this.count--;
				res = true;
				continue;
			}
		}
        if (res == false)
        {
            System.out.format("Not found repository with id: %s\n", id);

        }
        else
        {
            System.out.println("Done!");

        }
        return res;
    }

    public Repository FindItem(String id)
    {
        System.out.println("Find the repository");

        Repository res = null;
        for(Repository rep: repositoryList)
        {
        	if(rep.getId() == id) {
        		res = rep;
        	}
        }
        if (res == null)
        {
            System.out.format("Not found repository with id: %s\n", id);

        }
        else
        {
            System.out.println("Done!");
        }
        return res;
    }

    public boolean Clear()
    {
        System.out.println("\nClear all repositories");

        repositoryList.clear();

        return true;

    }
    // Overriding
}
